### PR TITLE
ci(.github): fix pr-comments action

### PR DESCRIPTION
**Pull request by lahabana**
### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .github/workflows/pr-comments.yaml [hunk 1]
```diff
@@ -14,28 +14,26 @@ jobs:
...
         env:
-          REPO: ${{ github.repository }}
-          PR_NO: ${{ github.event.issue.number }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
-          repository: ${{ steps.get-branch.outputs.repo }}
-          ref: ${{ steps.get-branch.outputs.branch }}
+          repository: ${{ env.REPO }}
```
Comment (lukidzi) [inline .github/workflows/pr-comments.yaml:35]: Does it work? I cannot find the definition of env

Comment (lukidzi) [reply .github/workflows/pr-comments.yaml:35]: Oh, here ```echo "BRANCH_NAME=$(gh pr view ${{ github.event.issue.number }} --repo ${{ github.repository }} --json headRefName --jq '.headRefName')" >> $GITHUB_ENV```

Comment (lahabana) [reply .github/workflows/pr-comments.yaml:35]: Yes exactly it's now the recommended way to pass info between steps in actions see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

**Comment by slonka** [comment]
does this mean I'll be able to do `/golden_files` etc. now?

**Pull request merged**